### PR TITLE
cli test subnet u2p

### PR DESCRIPTION
**Pull request by tstrych**
`================== 21 passed, 18 skipped in 327.16s (0:05:27) ==================`

please @rdrazny look at the tests, 
Important thing is that I added at the begining `:CaseImportance: Medium` 
Because we require this information for the tests.  
If you want to change it please let me know.

### CODE REVIEW

**Comment by tstrych** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_subnet.py

File: tests/foreman/cli/test_subnet.py [hunk 1]
```diff
@@ -69,532 +72,537 @@ def invalid_missing_attributes():
...
+    assert ipam_type in subnet['ipam']
+
+    # delete subnet
+    Subnet.delete({'id': subnet['id']})
+    with pytest.raises(CLIReturnCodeError):
+        Subnet.info({'id': subnet['id']})
+
+
+@pytest.mark.tier2
+@pytest.mark.parametrize('options', **parametrized(invalid_missing_attributes()))
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_subnet.py:157]: Make sure parameterized tests have their docstrings updated to indicate they are.

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_subnet.py [hunk 2]
```diff
@@ -69,532 +72,537 @@ def invalid_missing_attributes():
...
+
+@pytest.mark.tier2
+@pytest.mark.upgrade
+@pytest.mark.parametrize('pool', **parametrized(invalid_addr_pools()))
+def test_negative_create_with_address_pool(pool):
+    """Create subnet with invalid address pool range
+
+    :id: c7824327-b5ef-4f95-bd4b-ba4eff73551c
+
+    :expectedresults: Subnet is not created
```
Comment (jyejare) [inline tests/foreman/cli/test_subnet.py:179]: Add parametrized token in docstring.

File: tests/foreman/cli/test_subnet.py [hunk 3]
```diff
@@ -69,532 +72,537 @@ def invalid_missing_attributes():
...
+
+
+@pytest.mark.tier2
+@pytest.mark.parametrize('options', **parametrized(invalid_missing_attributes()))
+def test_negative_create_with_attributes(options):
+    """Create subnet with invalid or missing required attributes
+
+    :id: de468dd3-7ba8-463e-881a-fd1cb3cfc7b6
+
+    :expectedresults: Subnet is not created
```
Comment (jyejare) [inline tests/foreman/cli/test_subnet.py:163]: Add parametrized token in docstring.

File: tests/foreman/cli/test_subnet.py [hunk 4]
```diff
@@ -69,532 +72,537 @@ def invalid_missing_attributes():
...
-        """Presence of set parameter option in command
 
-        :id: f6ef75b2-6932-460c-80ba-2745244ec244
+@pytest.mark.tier2
+@pytest.mark.parametrize('options', **parametrized(invalid_missing_attributes()))
+def test_negative_update_attributes(options):
+    """Update subnet with invalid or missing required attributes
 
-        :steps:
+    :id: ab60372e-cef7-4495-bd66-68e7dbece475
```
Comment (jyejare) [inline tests/foreman/cli/test_subnet.py:199]: Add parametrized token in docstring.

File: tests/foreman/cli/test_subnet.py [hunk 5]
```diff
@@ -69,532 +72,537 @@ def invalid_missing_attributes():
...
-    @pytest.mark.tier1
-    def test_positive_create_with_parameter(self):
-        """Subnet with parameters can be created
+@pytest.mark.tier2
+@pytest.mark.parametrize('options', **parametrized(invalid_addr_pools()))
+def test_negative_update_address_pool(options):
+    """Update subnet with invalid address pool
 
-        :id: cf40a3f7-5a09-41aa-8b48-874a2af7057d
+    :id: d0a857b4-be10-4b5d-86d4-43cf99c11619
```
Comment (jyejare) [inline tests/foreman/cli/test_subnet.py:220]: Add parametrized token in docstring.

File: tests/foreman/cli/test_subnet.py [hunk 6]
```diff
@@ -69,532 +72,537 @@ def invalid_missing_attributes():
...
+    """Presence of set parameter option in command
 
-    @pytest.mark.stubbed
-    @pytest.mark.tier1
-    def test_positive_create_with_parameter_and_multiple_values(self):
-        """Subnet parameters can be created with multiple values
+    :id: f6ef75b2-6932-460c-80ba-2745244ec244
 
-        :id: 1e9ef184-bdf9-4eba-8055-f55b0dd9d6a0
+    :steps:
```
Comment (jyejare) [inline tests/foreman/cli/test_subnet.py:246]:
```suggestion
    :steps: Check if 'set-parameter' option is present in subnet command.
```

File: tests/foreman/cli/test_subnet.py [hunk 7]
```diff
@@ -69,532 +72,537 @@ def invalid_missing_attributes():
...
 
-        :steps:
+        1. Check if 'set-parameter' option is present in
```
Comment (jyejare) [inline tests/foreman/cli/test_subnet.py:248]: Need not to enumerate if we have only one step.

**Review by jyejare — CHANGES_REQUESTED** [review]
ACK pending parameterized tokens.

**Comment by tstrych** [comment]
thanks for reviews ;)

File: tests/foreman/cli/test_subnet.py [hunk 8]
```diff
@@ -69,532 +72,537 @@ def invalid_missing_attributes():
...
+    :CaseImportance: Critical
+    """
+    name = gen_choice(list(valid_data_list().values()))
+    pool = sorted(valid_addr_pools()[0])
+    mask = '255.255.255.0'
+    # generate pool range from network address
+    network = gen_ipaddr()
+    from_ip = re.sub(r'\d+$', str(pool[0]), network)
+    to_ip = re.sub(r'\d+$', str(pool[1]), network)
+    domains_amount = random.randint(3, 5)
```
[the remark this answers is not in the document]
Comment (tstrych) [reply tests/foreman/cli/test_subnet.py:93]: sure :+1: 
these tests are not mine, so don't want to changes in the logic of the tests, but this is really small change which brings some speedup, will implement.

**Comment by JacobCallahan** [comment]
trigger: test-robottelo
pytest: -n 3 --collect-only tests/foreman/cli/test_subnet.py

File: tests/foreman/cli/test_subnet.py [hunk 9]
```diff
@@ -69,532 +72,542 @@ def invalid_missing_attributes():
...
+    """Presence of set parameter option in command
 
-    @pytest.mark.stubbed
-    @pytest.mark.tier1
-    def test_positive_create_with_parameter_and_multiple_values(self):
-        """Subnet parameters can be created with multiple values
+    :id: f6ef75b2-6932-460c-80ba-2745244ec244
 
-        :id: 1e9ef184-bdf9-4eba-8055-f55b0dd9d6a0
+    :steps: 1. Check if 'set-parameter' option is present in subnet command.
```
Comment (jyejare) [inline tests/foreman/cli/test_subnet.py:254]: Even remove 1 there, keep it plain.

Non blocking comment.

**Review by jyejare — APPROVED** [review]
ACK!

**Comment by tstrych** [comment]
trigger: test-robottelo
pytest: --collect-only tests/foreman/cli/test_subnet.py

**Comment by tstrych** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_subnet.py

**Comment by tstrych** [comment]
Okay, now I can see the pr-tester is disabled. and run 68 failed at the beginning at command parsing. 
Precisely ` mapping values are not allowed here` 

So problem not connected.

Result locally:
```
pytest --collect-only tests/foreman/cli/test_subnet.py
============================================================================ test session starts ============================================================================
collected 39 items                                                                                                                                                          

<Package cli>
  <Module test_subnet.py>
    <Function test_positive_CRUD>
    <Function test_negative_create_with_attributes[0]>
    <Function test_negative_create_with_attributes[1]>
    <Function test_negative_create_with_attributes[2]>
    <Function test_negative_create_with_attributes[3]>
    <Function test_negative_create_with_attributes[4]>
    <Function test_negative_create_with_attributes[5]>
    <Function test_negative_create_with_address_pool[0]>
    <Function test_negative_create_with_address_pool[1]>
    <Function test_negative_create_with_address_pool[2]>
    <Function test_negative_create_with_address_pool[3]>
    <Function test_negative_update_attributes[0]>
    <Function test_negative_update_attributes[1]>
    <Function test_negative_update_attributes[2]>
    <Function test_negative_update_attributes[3]>
    <Function test_negative_update_attributes[4]>
    <Function test_negative_update_attributes[5]>
    <Function test_negative_update_address_pool[0]>
    <Function test_negative_update_address_pool[1]>
    <Function test_negative_update_address_pool[2]>
    <Function test_negative_update_address_pool[3]>
    <Function test_positive_set_parameter_option_presence>
    <Function test_positive_create_with_parameter>
    <Function test_positive_create_with_parameter_and_multiple_values>
    <Function test_positive_create_with_parameter_and_multiple_names>
    <Function test_negative_create_with_parameter_and_invalid_separator>
    <Function test_positive_create_with_multiple_parameters>
    <Function test_negative_create_with_duplicated_parameters>
    <Function test_positive_inherit_subnet_parmeters_in_host>
    <Function test_negative_inherit_subnet_parmeters_in_host>
    <Function test_positive_subnet_parameters_override_from_host>
    <Function test_positive_subnet_parameters_override_impact_on_subnet>
    <Function test_positive_update_parameter>
    <Function test_negative_update_parameter>
    <Function test_positive_update_subnet_parameter_host_impact>
    <Function test_positive_delete_subnet_parameter>
    <Function test_positive_delete_multiple_parameters>
    <Function test_positive_delete_subnet_parameter_host_impact>
    <Function test_positive_delete_subnet_parameter_overrided_host_impact>

======================================================================== 39 tests collected in 0.05s ========================================================================
```

**Comment by tstrych** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_subnet.py

**Review by mshriver — APPROVED** [review]

**Pull request merged**